op-chain-ops/deployer/state/deploy_config.go

```diff
@@ -0,0 +1,154 @@
+				GovernanceTokenSymbol: "OP",
+				GovernanceTokenName:   "Optimism",
+			},
+			GasPriceOracleDeployConfig: genesis.GasPriceOracleDeployConfig{
+				GasPriceOracleBaseFeeScalar:     1368,
+				GasPriceOracleBlobBaseFeeScalar: 810949,
+			},
+			EIP1559DeployConfig: genesis.EIP1559DeployConfig{
+				EIP1559Denominator:       50,
+				EIP1559DenominatorCanyon: 250,
+				EIP1559Elasticity:        6,
+			},
+			UpgradeScheduleDeployConfig: genesis.UpgradeScheduleDeployConfig{
+				L2GenesisRegolithTimeOffset: u64UtilPtr(0),
+				L2GenesisCanyonTimeOffset:   u64UtilPtr(0),
+				L2GenesisDeltaTimeOffset:    u64UtilPtr(0),
+				L2GenesisEcotoneTimeOffset:  u64UtilPtr(0),
+				L2GenesisFjordTimeOffset:    u64UtilPtr(0),
+				L2GenesisGraniteTimeOffset:  u64UtilPtr(0),
+				UseInterop:                  false,
+			},
+			L2CoreDeployConfig: genesis.L2CoreDeployConfig{
+				L2BlockTime:               2,
+				FinalizationPeriodSeconds: 12,
+				MaxSequencerDrift:         600,
+				SequencerWindowSize:       3600,
+				ChannelTimeoutBedrock:     300,
+				SystemConfigStartBlock:    0,
+			},
+		},
+	}
+}
+
+func CombineDeployConfig(intent *Intent, chainIntent *ChainIntent, state *State, chainState *ChainState) (genesis.DeployConfig, error) {
+	cfg := DefaultDeployConfig()
+
+	var err error
+	if len(intent.GlobalDeployOverrides) > 0 {
+		cfg, err = mergeJSON(cfg, intent.GlobalDeployOverrides)
+		if err != nil {
+			return genesis.DeployConfig{}, fmt.Errorf("error merging global L2 overrides: %w", err)
+
+		}
+	}
+
+	if len(chainIntent.DeployOverrides) > 0 {
+		cfg, err = mergeJSON(cfg, chainIntent.DeployOverrides)
+		if err != nil {
+			return genesis.DeployConfig{}, fmt.Errorf("error merging chain L2 overrides: %w", err)
+		}
+	}
+
+	cfg.L2ChainID = chainState.ID.Big().Uint64()
+	cfg.L1DependenciesConfig = genesis.L1DependenciesConfig{
+		L1StandardBridgeProxy:       chainState.L1StandardBridgeProxyAddress,
+		L1CrossDomainMessengerProxy: chainState.L1CrossDomainMessengerProxyAddress,
+		L1ERC721BridgeProxy:         chainState.L1ERC721BridgeProxyAddress,
+		SystemConfigProxy:           chainState.SystemConfigProxyAddress,
+		OptimismPortalProxy:         chainState.OptimismPortalProxyAddress,
+		ProtocolVersionsProxy:       state.SuperchainDeployment.ProtocolVersionsProxyAddress,
+	}
+	cfg.OperatorDeployConfig = genesis.OperatorDeployConfig{
+		BatchSenderAddress:  chainIntent.Roles.Batcher,
+		P2PSequencerAddress: chainIntent.Roles.UnsafeBlockSigner,
+	}
+	cfg.BatchInboxAddress = calculateBatchInboxAddr(chainState.ID)
+	cfg.L1ChainID = intent.L1ChainID
+
+	return cfg, nil
+}
+
+func mergeJSON[T any](in T, overrides ...map[string]any) (T, error) {
+	var out T
+	inJSON, err := json.Marshal(in)
+	if err != nil {
+		return out, err
+	}
+
+	var tmpMap map[string]interface{}
+	if err := json.Unmarshal(inJSON, &tmpMap); err != nil {
+		return out, err
+	}
+
+	for _, override := range overrides {
+		for k, v := range override {
+			tmpMap[k] = v
```

Comment:
If there are multiple `overrides` passed as arguments into the `mergeJSON` function and they share a `key` it looks like the last one will take precedence, i.e. it would overwrite previous ones on this line. Is that a situation that could happen? If so, might be good to note this in the function comment or return an error here if we want to disallow that to happen

Reply:
This is a desired behavior of the method - we need to merge multiple overrides from different levels of the state file.

Reply:
Added a comment describing the behavior.